tests/ui/AttachResult.spec.ts

```diff
@@ -0,0 +1,237 @@
+import { AttachResult, IAttachResultOptions } from "../../src/ui/AttachResult/AttachResult";
+import { Mock, Fake, Simulate } from "coveo-search-ui-tests";
+import { $$, InitializationEvents, QueryEvents, IBuildingQueryEventArgs, IQueryResult } from "coveo-search-ui";
+
+describe("AttachResult", () => {
+    let attachResult: Mock.IBasicComponentSetup<AttachResult>;
+    let fakeResult = Fake.createFakeResult();
+
+    beforeEach(() => {
+        attachResult = Mock.basicResultComponentSetup<AttachResult>(AttachResult);
+    });
+
+    afterEach(() => {
+        attachResult = null;
+    });
+
+    it("should have two children elements", () => {
+        expect(attachResult.cmp.element.childNodes.length).toBe(2);
+        expect(attachResult.cmp.element.querySelector(".coveo-caption-for-icon")).not.toBeNull();
+        expect(attachResult.cmp.element.querySelector(".coveo-icon-attach")).not.toBeNull();
+    });
+
+    describe("when configuring isAttached", () => {
+        let faker;
+        let isAttachedSpy : jasmine.Spy;
+        beforeEach(() => {
+            faker = {
+                isAttachedMock: function(result : IQueryResult) {
+                    return Promise.resolve(true);
+                }
+            };
+
+            isAttachedSpy = spyOn(faker, "isAttachedMock").and.callThrough();
+        });
+
+        it("should call isAttached on initialize", () => {
+            attachResult = Mock.optionsResultComponentSetup(AttachResult, { isAttached: faker.isAttachedMock }, fakeResult);
+            expect(faker.isAttachedMock).toHaveBeenCalledWith(fakeResult);
+        });
+
+        it("should be attached when isAttached returns true", (done) => {
+            isAttachedSpy.and.returnValue(Promise.resolve(true));
+            attachResult = Mock.optionsResultComponentSetup(AttachResult, { isAttached: faker.isAttachedMock }, fakeResult);
+            setTimeout(() => {
```

Comment:
Seems strange that these `setTimeout` are needed?

Reply:
To simulate the back and forth of a server call?

Reply:
i was having a hard time trying to make the test work, for some reason I cannot call `done()` in the spy `.and.callFake`, the test kept timeouting.. do you have an idea why?

Reply:
Hmm, I've run into problems when using promises with the jest framework. There I solve it by moving the async method into the `beforeEach`. If the promise is the issue, maybe there is a thread that discusses it for Jasmine?